### PR TITLE
Add `manage_slm` and `read_slm` cluster privileges

### PR DESCRIPTION
**Pull request by dakrone**
This adds two more built in roles -

`manage_slm` which has permission to perform any of the SLM actions, as well as
stopping, starting, and retrieving the operation status of ILM.

`read_slm` which has permission to retrieve snapshot lifecycle policies as well
as retrieving the operation status of ILM.

Relates to #38461

### CODE REVIEW

**Comment by dakrone** [comment]
I'll add in to the documentation that these roles are required for certain operations once this an #41510 are merged.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilege.java [hunk 1]
```diff
@@ -60,6 +63,9 @@
     private static final Automaton READ_CCR_AUTOMATON = patterns(ClusterStateAction.NAME, HasPrivilegesAction.NAME);
     private static final Automaton MANAGE_ILM_AUTOMATON = patterns("cluster:admin/ilm/*");
     private static final Automaton READ_ILM_AUTOMATON = patterns(GetLifecycleAction.NAME, GetStatusAction.NAME);
+    private static final Automaton MANAGE_SLM_AUTOMATON =
+        patterns("cluster:admin/slm/*", StartILMAction.NAME, StopILMAction.NAME, GetStatusAction.NAME);
```
Comment (gwbrown) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilege.java]: I understand why you can start/stop ILM with `manage_slm` but it's kind of awkward, huh.

**Review by AthenaEryma — APPROVED** [review]
LGTM with my relatively-minimal knowledge of our security framework.

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilege.java (hunk 1 above)
Comment (dakrone) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/privilege/ClusterPrivilege.java]: Yeah, gotta be able to stop/start it

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/1

**Pull request merged**